homeassistant/components/iaqualink/__init__.py

```diff
@@ -101,3 +116,36 @@
     hass.data[DOMAIN].clear()
 
     return all(await asyncio.gather(*tasks))
+
+
+def refresh_system(func):
+    """Force update all entities after state change."""
+
+    @wraps(func)
+    async def wrapper(self, *args, **kwargs):
+        """Call decorated function and send update signal to all entities."""
+        await func(self, *args, **kwargs)
+        async_dispatcher_send(self.hass, DOMAIN)
+
+    return wrapper
+
+
+class AqualinkEntity(Entity):
+    """Abstract class for all Aqualink platforms."""
+
+    async def async_added_to_hass(self) -> None:
+        """Set up a listener when this entity is added to HA."""
+        async_dispatcher_connect(self.hass, DOMAIN, self._update_callback)
+
+    @callback
+    def _update_callback(self) -> None:
+        self.async_schedule_update_ha_state(force_refresh=True)
```

Comment:
You should not do this

Reply:
Why not? We use this pattern in many places.